src/structures/command/parameters/types/UserParameter.js

```diff
@@ -13,31 +13,38 @@ module.exports = class UserParameter extends Parameter {
       acceptUser: defVal(options, 'acceptUser', true),
       acceptDeveloper: defVal(options, 'acceptDeveloper', true),
       acceptSelf: !!options.acceptSelf,
+      acceptPartial: !!options.acceptPartial,
       errors: {
         invalidUser: 'errors:invalidUser',
         acceptSelf: 'errors:sameUser',
         acceptBot: 'errors:invalidUserBot',
         acceptUser: 'errors:invalidUserNotBot',
         acceptDeveloper: 'errors:userCantBeDeveloper',
+        acceptPartial: 'errors:userPartial',
         ...(options.errors || {})
       }
     }
   }
 
-  static parse (arg, { t, client, author, guild }) {
+  static async parse (arg, { t, client, author, guild }) {
     if (!arg) return
 
     const regexResult = MENTION_REGEX.exec(arg)
     const id = regexResult && regexResult[1]
     const findMember = guild.members.cache.get(id) || guild.members.cache.find(m => m.user.username.toLowerCase().includes(arg.toLowerCase()) || m.displayName.toLowerCase().includes(arg.toLowerCase()))
+    try {
+      const partialUser = await client.users.fetch(id)

```

Comment:
It should only attempt to fetch if `this.acceptPartial` is true.
And you should be `.catch()`'ing it in case `id` is not a valid user snowflake.

```suggestion
      const partialUser = this.acceptPartial && await client.users.fetch(id).catch(() => null)

```